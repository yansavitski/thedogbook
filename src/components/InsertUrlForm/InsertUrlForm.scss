@import 'src/global';

.InsertUrlForm {
  display: flex;

  &-input {
    background: transparent;
    border: 0;
    color: $textColor;
    padding: 0;
    width: 0;

    @include transition(width);

    &:focus {
      outline: none;
    }

    &_show {
      border: 1px solid $borderColor;
      border-bottom-left-radius: $borderRadius;
      border-right: none;
      border-top-left-radius: $borderRadius;
      padding: $baseOffset/2;
      width: 200px;
    }
  }

  &-submitButton {
    &#{&}_show {
      border-bottom-left-radius: 0;
      border-top-left-radius: 0;
    }
  }
}
