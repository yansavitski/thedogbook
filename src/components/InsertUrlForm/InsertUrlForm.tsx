import './InsertUrlForm.scss'

import classNames from 'classnames'
import { Button } from 'components/Button/Button'
import React, { ChangeEvent, useCallback, useState } from 'react'

interface InsertUrlFormProps {
  className?: string
  disabled?: boolean
  onSubmit: (text: string) => void
}

export const InsertUrlForm: React.FunctionComponent<InsertUrlFormProps> = ({
  className,
  disabled = false,
  onSubmit,
}: InsertUrlFormProps) => {
  const [showInput, setShowInput] = useState<boolean>(false)
  const [text, setText] = useState<string>('')
  const hasText = Boolean(text)
  const handleMouseEnter = useCallback(() => {
    setShowInput(true)
  }, [setShowInput])
  const handleMouseLeave = useCallback(() => {
    if (!hasText) {
      setShowInput(false)
    }
  }, [text, setShowInput])
  const handleInputChange = useCallback(
    (event: ChangeEvent<HTMLInputElement>) => {
      setText(event.target.value)
    },
    [setText],
  )
  const handleSubmit = useCallback(() => {
    if (hasText) {
      onSubmit(text)
      setText('')
    }
  }, [text, setText, onSubmit])

  return (
    <form
      className={classNames('InsertUrlForm', className)}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
    >
      <input
        className={classNames(
          'InsertUrlForm-input',
          showInput && 'InsertUrlForm-input_show',
        )}
        type="text"
        placeholder="Please insert url to image"
        disabled={disabled}
        onChange={handleInputChange}
        value={text}
      />

      <Button
        className={classNames(
          'InsertUrlForm-submitButton',
          showInput && 'InsertUrlForm-submitButton_show',
        )}
        onClick={handleSubmit}
      >
        Insert URL to image
      </Button>
    </form>
  )
}
