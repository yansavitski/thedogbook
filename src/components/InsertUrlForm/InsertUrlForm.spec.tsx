import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import { InsertUrlForm } from './InsertUrlForm'

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

describe('<InsertUrlForm />', () => {
  it('renders correctly', () => {
    expect.assertions(1)
    const tree = shallow(<InsertUrlForm onSubmit={(): void => {}} />, {
      wrappingComponent: Provider,
      wrappingComponentProps: { store: mockStore() },
    })

    expect(toJson(tree)).toMatchSnapshot()
  })

  it('renders correctly with optional props', () => {
    expect.assertions(1)
    const tree = shallow(
      <InsertUrlForm
        className="customClassWrapper"
        onSubmit={(): void => {}}
        disabled
      />,
    )

    expect(toJson(tree)).toMatchSnapshot()
  })
})
