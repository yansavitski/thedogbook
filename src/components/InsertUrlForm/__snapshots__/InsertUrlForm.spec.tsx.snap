// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InsertUrlForm /> renders correctly 1`] = `
<form
  className="InsertUrlForm"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <input
    className="InsertUrlForm-input"
    disabled={false}
    onChange={[Function]}
    placeholder="Please insert url to image"
    type="text"
    value=""
  />
  <Component
    className=""
    onClick={[Function]}
  >
    Insert URL to image
  </Component>
</form>
`;

exports[`<InsertUrlForm /> renders correctly with optional props 1`] = `
<form
  className="InsertUrlForm customClassWrapper"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <input
    className="InsertUrlForm-input"
    disabled={true}
    onChange={[Function]}
    placeholder="Please insert url to image"
    type="text"
    value=""
  />
  <Component
    className=""
    onClick={[Function]}
  >
    Insert URL to image
  </Component>
</form>
`;
